### PR TITLE
SQL: add "validate.properties" property to JDBC's allowed list of settings

### LINKED ISSUE #38068: throw unknown parameter [serialization.format] when spark create elasticsearch jdbc connection
configuration: 
```
jdbcUrl: jdbc:es://http://192.168.1.81:9200
driverClass: org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcDriver
```
spark create ***elasticsearch jdbc*** connection throw follow exception, but ignore unknown parameter when use ***mysql jdbc*** or ***postgresql jdbc***
```
Exception in thread "main" org.elasticsearch.xpack.sql.jdbc.JdbcSQLException: Unknown parameter [serialization.format] ; did you mean []
	at org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcConfiguration.create(JdbcConfiguration.java:95)
	at org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcDriver.initCfg(JdbcDriver.java:82)
	at org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcDriver.connect(JdbcDriver.java:76)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anonfun$createConnectionFactory$1.apply(JdbcUtils.scala:59)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anonfun$createConnectionFactory$1.apply(JdbcUtils.scala:50)
	at org.apache.spark.sql.execution.datasources.jdbc.JDBCRDD$.resolveTable(JDBCRDD.scala:58)
	at org.apache.spark.sql.execution.datasources.jdbc.JDBCRelation.<init>(JDBCRelation.scala:113)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcRelationProvider.createRelation(JdbcRelationProvider.scala:45)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:335)
	at org.apache.spark.sql.hive.ConnectorCatalog$$anon$1.load(ConnectorCatalog.scala:108)
	at org.apache.spark.sql.hive.ConnectorCatalog$$anon$1.load(ConnectorCatalog.scala:74)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3599)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2379)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2342)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2257)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4000)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4004)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4874)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4880)
	at com.google.common.cache.LocalCache$LocalLoadingCache.apply(LocalCache.java:4898)
	at org.apache.spark.sql.hive.SnappyStoreHiveCatalog.getCachedHiveTable(SnappyStoreHiveCatalog.scala:312)
	at org.apache.spark.sql.hive.SnappyStoreHiveCatalog.lookupRelation(SnappyStoreHiveCatalog.scala:654)
	at org.apache.spark.sql.hive.SnappyStoreHiveCatalog.lookupRelation(SnappyStoreHiveCatalog.scala:738)
	at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.org$apache$spark$sql$catalyst$analysis$Analyzer$ResolveRelations$$lookupTableFromCatalog(Analyzer.scala:457)
	at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$8.applyOrElse(Analyzer.scala:479)
	at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$$anonfun$apply$8.applyOrElse(Analyzer.scala:464)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$resolveOperators$1.apply(LogicalPlan.scala:61)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$resolveOperators$1.apply(LogicalPlan.scala:61)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:70)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:60)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$4.apply(TreeNode.scala:307)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:188)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:305)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$4.apply(TreeNode.scala:307)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:188)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:305)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$4.apply(TreeNode.scala:307)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:188)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:305)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan$$anonfun$1.apply(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$4.apply(TreeNode.scala:307)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:188)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:305)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:58)
	at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:464)
	at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveRelations$.apply(Analyzer.scala:454)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor$$anonfun$execute$1$$anonfun$apply$1.apply(RuleExecutor.scala:85)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor$$anonfun$execute$1$$anonfun$apply$1.apply(RuleExecutor.scala:82)
	at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
	at scala.collection.immutable.List.foldLeft(List.scala:84)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor$$anonfun$execute$1.apply(RuleExecutor.scala:82)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor$$anonfun$execute$1.apply(RuleExecutor.scala:74)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:74)
	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:69)
	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:67)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:50)
	at org.apache.spark.sql.SnappySession.executePlan(SnappySession.scala:228)
	at org.apache.spark.sql.SnappySession$.sqlPlan(SnappySession.scala:2257)
	at org.apache.spark.sql.SnappySession$$anonfun$sql$1.apply(SnappySession.scala:203)
	at org.apache.spark.sql.SnappySession$$anonfun$sql$1.apply(SnappySession.scala:203)
	at org.apache.spark.sql.aqp.SnappyContextFunctions.sql(SnappyContextFunctions.scala:110)
	at org.apache.spark.sql.SnappySession.sql(SnappySession.scala:203)
	at org.apache.spark.sql.SnappySession.sql(SnappySession.scala:79)
	at org.apache.spark.sql.SQLContext.sql(SQLContext.scala:700)
Caused by: org.elasticsearch.xpack.sql.client.ClientException: Unknown parameter [serialization.format] ; did you mean []
	at org.elasticsearch.xpack.sql.client.ConnectionConfiguration.checkPropertyNames(ConnectionConfiguration.java:153)
	at org.elasticsearch.xpack.sql.client.ConnectionConfiguration.<init>(ConnectionConfiguration.java:88)
	at org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcConfiguration.<init>(JdbcConfiguration.java:148)
	at org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcConfiguration.create(JdbcConfiguration.java:91)
	... 77 more
```

### PR DESCRIPTION
**Pull request by astefan**
This PR adds an additional property to JDBC, to be specified in URL or in a `Properties` instance, that allows an user to disable the default behavior of checking and throwing an exception in case of unrecognizable properties found in either URL or `Properties` object.

Fixes https://github.com/elastic/elasticsearch/issues/38068.

### CODE REVIEW

File: x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java [hunk 1]
```diff
@@ -113,6 +117,66 @@ public void testHttpWithSSLDisabledFromPropertyAndEnabledFromProtocol() throws E
...
+    public void testValidateProperties() {
+        Exception e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://test:9200?pagee.size=12&validate.properties=true"));
+        assertEquals("Unknown parameter [pagee.size]; did you mean [page.size]", e.getMessage());
+        
+        e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://test:9200?&validate.properties=true&something=some_value"));
+        assertEquals("Unknown parameter [something]; did you mean []", e.getMessage());
+        
+        Properties properties  = new Properties();
+        properties.setProperty(PROPERTIES_VALIDATION, "true");
+        e = expectThrows(JdbcSQLException.class, () -> JdbcConfiguration.create("jdbc:es://test:9200?something=some_value", properties, 0));
```
Comment (matriv) [inline x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java]: This seems like a test for `testValidatePropertiesDefault`, did you miss to add the `validate.properties=true`?

**Review by matriv — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java]: @matriv the properties can be specified in a `Properties` object, as well. And in this case, it's specified like this (see above  `properties.setProperty(PROPERTIES_VALIDATION, "true");`).

Comment (matriv) [reply x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java]: Oh, sorry, totally missed that somehow :-)

**Pull request merged**